rom/dev/src/flow/cold_reset/fw_processor.rs

```diff
@@ -410,4 +461,105 @@ impl FirmwareProcessor {
 
         (nb, nf)
     }
+
+    /// Read measurement from mailbox and extends it into PCR31
+    ///
+    /// # Arguments
+    /// * `pcr_bank` - PCR Bank
+    /// * `sha384` - SHA384
+    /// * `txn` - Mailbox Receive Transaction
+    ///
+    /// # Returns
+    /// * `()` - Ok
+    ///     Err - StashMeasurementReadFailure
+    fn stash_measurement(
+        pcr_bank: &mut PcrBank,
+        sha384: &mut Sha384,
+        txn: &mut MailboxRecvTxn,
+        log_index: u32,
+    ) -> CaliptraResult<()> {
+        const MEASUREMENT_WORDS: usize =
+            core::mem::size_of::<StashMeasurement>() / core::mem::size_of::<u32>();
+        let mut slice: [u32; MEASUREMENT_WORDS] = [0u32; MEASUREMENT_WORDS];
+        txn.copy_request(&mut slice)?;
+
+        let measurement = StashMeasurement::read_from(slice.as_bytes())
+            .ok_or(CaliptraError::FW_PROC_STASH_MEASUREMENT_READ_FAILURE)?;
+
+        // Extend measurement into PCR31.
+        Self::extend_measurement(
+            pcr_bank,
+            sha384,
+            measurement.measurement.as_bytes(),
+            log_index as usize,
+        )?;
+
+        Ok(())
+    }
+
+    /// Extends measurement into PCR31 and logs it to PCR log.
+    ///
+    /// # Arguments
+    /// * `pcr_bank` - PCR Bank
+    /// * `sha384` - SHA384
+    /// * `measurement` - Measurement
+    /// * `log_index` - Log index
+    ///
+    /// # Returns
+    /// * `()` - Ok
+    ///    Error code on failure.
+    fn extend_measurement(
+        pcr_bank: &mut PcrBank,
+        sha384: &mut Sha384,
+        measurement: &[u8],
+        log_index: usize,
+    ) -> CaliptraResult<()> {
+        // Extend measurement into PCR31.
+        pcr_bank.extend_pcr(PCR_ID_STASH_MEASUREMENT, sha384, measurement.as_bytes())?;
+
+        // Log measurement to the PCR log.
+        Self::log_measurement(measurement.as_bytes(), log_index)?;
+
+        Ok(())
+    }
+
+    /// Log mesaure data to the Stash Measurement log
+    ///
+    /// # Arguments
+    /// * `data` - PCR data
+    /// * `log_index` - Log index
+    ///
+    /// # Return Value
+    /// * `Ok(())` - Success
+    /// * `Err(GlobalErr::PcrLogUpsupportedDataLength)` - Unsupported data length
+    /// * `Err(GlobalErr::MeasurementLogExhausted)` - Measurement log exhausted
+    ///
+    pub fn log_measurement(data: &[u8], log_index: usize) -> CaliptraResult<()> {
+        if data.len() > 48 {
+            return Err(CaliptraError::ROM_GLOBAL_PCR_LOG_UNSUPPORTED_DATA_LENGTH);
+        }
+
+        if (log_index * core::mem::size_of::<PcrLogEntry>()) > MEASUREMENT_LOG_SIZE {
+            return Err(CaliptraError::ROM_GLOBAL_MEASUREMENT_LOG_EXHAUSTED);
+        }
+
+        // Create a PCR log entry
+        let mut pcr_log_entry = PcrLogEntry {
+            id: PcrLogEntryId::StashMeasurement as u16,
```

Comment:
should the log entry also contain the metadata?

Reply:
I changed the reserved1 field in the PcrLogEntry to 'metadata'. The metadata from the measurement is stored in this field in the PCR log; it won't be extended to PCR31 though. Will this suffice?